modules/low-watermark/src/test/java/org/apache/ignite/internal/lowwatermark/LowWatermarkImplTest.java

```diff
@@ -336,6 +342,36 @@ void testUpdateAndNotifyNotInvokeFailureManagerWhenGetNodeStoppingException() {
         verify(failureManager, never()).process(any());
     }
 
+    @Test
+    void testParallelScheduleUpdates() throws Exception {
+        assertThat(lowWatermarkConfig.updateIntervalMillis().update(300L), willCompleteSuccessfully());
+
+        assertThat(lowWatermark.startAsync(new ComponentContext()), willCompleteSuccessfully());
+
+        runRace(
+                () -> lowWatermark.scheduleUpdates(),
+                () -> lowWatermark.scheduleUpdates(),
+                () -> lowWatermark.scheduleUpdates(),
+                () -> lowWatermark.scheduleUpdates()
+        );
+
+        Thread.sleep(1_000);
+
+        verify(lwmChangedListener, atLeast(2)).notify(any());
```

Comment:
Why is it at least 2? It seems that each subsequent task might cancel the previous one, so why can't it be just 1 (the last one)?

Reply:
Because the interval is 300ms, then after 1 second, regardless of the platform (well, I suppose so), at least 2 watermark updates will be performed, in normal systems it should of course be 3, but I don’t want a flaky test.

Reply:
But what if this thread gets delayed for some reason for another second? Will this result in more than 4 invocations?